app/server/api/worker_callback_api.py

```diff
@@ -0,0 +1,52 @@
+from flask import Blueprint, request, make_response, jsonify, g
+from flask.views import MethodView
+
+from server import db
+from server.utils.auth import requires_auth
+from server.models.credit_transfer import CreditTransfer
+from server.models.worker_messages import WorkerMessages
+
+from datetime import datetime
+import time
+worker_callback_blueprint = Blueprint('worker_callback', __name__)
+
+class WorkerCallbackAPI(MethodView):
+    @requires_auth(allowed_basic_auth_types=('internal'))
+    def post(self):
+        post_data = request.get_json()
+        blockchain_task_uuid = post_data.get('blockchain_task_uuid')
+        timestamp = datetime.fromtimestamp(post_data.get('timestamp'))
+        blockchain_status = post_data.get('blockchain_status')
+        error = post_data.get('error')
+        message = post_data.get('message')
+        blockchain_hash = post_data.get('hash')
+
+        transfer = CreditTransfer.query.execution_options(show_all=True).filter_by(blockchain_task_uuid = blockchain_task_uuid).first()
```

Comment:
Shouldn't this be a filter on something that also searches for matching `Exchange` objects too?

Reply:
@enjeyw Looking at this again, do we actually want the status fields in the `Exchange` model in the first place? Like when you look at the transfer list on the FE, the `blockchain_status` you see  for any given transfer is actually the `blockchain_status` of an exchange's `from_transfer` or `to_transfer`  object.

Reply:
yeah we still do want the status field - currently it's a case of the Front End not being sufficiently developed to support exchanges.

Reply:
Makes sense. I made it so the status field on _both_ are set now. The redundancy isn't ideal, but  I think it's worth it to have it shown in the front end until it the FE supports exchanges.